File: Laba2SSPR/src/main/java/com/Laba2/Laba2/core/Category.java
package com.Laba2.Laba2.core;

import javax.persistence.*;

@Entity
@Table(name = "category")
public class Category {
    @Id
    @GeneratedValue
    private int id;
    @Column(name = "name")
    private String name;
    @Column(name = "image")
    private String imgPath;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "ID_type")
    private TypeTechnika technika;

    public Category(){

    }

    public Category(String name, String imgPath,TypeTechnika technika){
        this.name = name;
        this.imgPath = imgPath;
        this.technika = technika;
    }

    public String getImgPath() {
        return this.imgPath;
    }

    public void setImgPath(String imgPath) {
        this.imgPath = imgPath;
    }

    public TypeTechnika getTechnika() {
        return this.technika;
    }

    public void setTechnika(TypeTechnika technika) {
        this.technika = technika;
    }

    public int getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }
}

File: Laba3SSPR/src/Entity/Category.php
<?php

namespace App\Entity;

use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\CategoryRepository")
 */
class Category
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    private $Name;

    /**
     * @ORM\Column(type="string", length=255, nullable=true)
     */
    private $ImagePath;

    /**
     * @ORM\ManyToOne(targetEntity="App\Entity\TypeTechnika", inversedBy="categories")
     * @ORM\JoinColumn(nullable=false)
     */
    private $Id_Type;

    /**
     * @ORM\OneToMany(targetEntity="App\Entity\Cars", mappedBy="Id_category")
     */
    private $cars;

    public function __construct()
    {
        $this->cars = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->Name;
    }

    public function setName(string $Name): self
    {
        $this->Name = $Name;

        return $this;
    }

    public function getImagePath(): ?string
    {
        return $this->ImagePath;
    }

    public function setImagePath(?string $ImagePath): self
    {
        $this->ImagePath = $ImagePath;

        return $this;
    }

    public function getIdType(): ?TypeTechnika
    {
        return $this->Id_Type;
    }

    public function setIdType(?TypeTechnika $Id_Type): self
    {
        $this->Id_Type = $Id_Type;

        return $this;
    }

    /**
     * @return Collection|Cars[]
     */
    public function getCars(): Collection
    {
        return $this->cars;
    }

    public function addCar(Cars $car): self
    {
        if (!$this->cars->contains($car)) {
            $this->cars[] = $car;
            $car->setIdCategory($this);
        }

        return $this;
    }

    public function removeCar(Cars $car): self
    {
        if ($this->cars->contains($car)) {
            $this->cars->removeElement($car);
            // set the owning side to null (unless already changed)
            if ($car->getIdCategory() === $this) {
                $car->setIdCategory(null);
            }
        }

        return $this;
    }

    public function __toString()
    {
        // TODO: Implement __toString() method.
        return $this->Name;
    }
}
